src/chapters/functions/studio.rst

```diff
@@ -0,0 +1,169 @@
+
+At the end, you will be able to create strings of shapes, like this nifty diamond:
+
+::
+
+       #     
+      ###    
+     #####   
+    #######  
+   #########   
+    #######    
+     #####     
+      ###      
+       #
+
+There is no starter code for this studio, so create a new Node.js project at `repl.it <https://repl.it/student>`_ to get started.
+
+Rectangles
+----------
+
+#. Write a function ``line(size)`` that returns a line with exactly ``size``
+   hashes. 
+
+   .. sourcecode:: js
+   
+      console.log(line(5));
+      
+   **Output:** 
+   
+   ::
+   
+      #####
+
+#. Write a function ``square(size)`` that returns an ``size`` by ``size`` square
+   of hashes. Use your ``line`` function to do this.
+
+   .. sourcecode:: js
+   
+      console.log(square(5));
+
+   **Output:**
+   
+   ::
+   
+      #####  
+      #####  
+      #####  
+      #####  
+      #####
+
+   .. tip:: The newline character, ``\n``, will be helpful to you.
+   
+   .. warning:: For this and all other functions in this studio, make sure you do NOT have a newline character at the end of your string. Not only will ``console.log`` add a newline there for you, but having an extra newline at the end will make life harder for you toward the end of the studio.
+   
+
+#. Write a function ``rectangle(width, height)`` that returns a
+   rectangle with the given width and height. Use your
+   ``line`` function to do this. 
+   
+   .. sourcecode:: js
+   
+      console.log(rectangle(5, 3));
+
+   **Output:** 
+   
+   ::
+   
+      #####  
+      #####  
+      #####
+      
+   Now, go back and rewrite ``square`` to use ``rectangle``.
+
+Triangles
+---------
+
+#. Write a function ``stairs(height)`` that prints the staircase pattern shown below,
+   with the given height. Use your ``line`` function to do this.
+   
+   .. sourcecode:: js
+   
+      console.log(stairs(5));
+      
+   **Output:**
+   
+   ::
+   
+      #  
+      ##  
+      ###  
+      ####  
+      #####
+
+#. Write a function ``spaceLine(numSpaces, numChars)`` that returns a line
+   with exactly the specified number of spaces, followed by the
+   specified number of hashes, followed again by ``numSpaces`` more spaces. 
+   
+   .. sourcecode:: js
+   
+      console.log(spaceLine(3, 5));
+      
+   **Output:**
+
+   ::
+   
+      ___#####___
+
+   .. note:: We have inserted underscores to represent spaces, so they are visible in the output. Don't do this in your code.
+
+#. Write a function ``triangle(height)`` that returns a triangle of
+   the given height. 
+   
+   .. sourcecode:: js
+   
+      console.log(triangle(5));
+      
+   **Output:**
+
+   ::
+   
+          #
+         ###
+        #####
+       #######
+      #########
+
+   .. tip:: Consider the top line of the triangle to be level 0, the next to be line 1, and so on. Then line ``i`` is a space-line with ``height - i - 1`` spaces and ``2 * i + 1`` hashes.
+
+Diamonds
```

Comment:
Students who get this will have no trouble finishing the entire studio in the allotted time.  Students who struggle will hit the wall around the isosceles triangle.

I'd classify "Diamonds" as a bonus mission, and merge the "use characters other than '#'" missions into a single task.  This could go above the diamonds task in the bonus missions, or it could be slipped in after the ``line`` task.

Reply:
Makes sense. I thought that it felt a little big. I'm happy to make diamonds a bonus mission.

Is there a better way to point students in the right direction on the triangle problem? Or do they just need to struggle with it?

Reply:
Struggle.  As the last task before the diamonds, it is well placed to push the students outside of their comfort zone, but not so far that the solution is out of reach.